quinn-h3/src/server.rs

```diff
@@ -197,9 +485,37 @@ impl From<quinn::Connecting> for Connecting {
     }
 }
 
+/// Stream of incoming requests for a connection.
+///
+/// This yeilds [`RecvRequest`] futures as soon as its underlying QUIC stream is open.
+///
+/// ```
+/// use futures::StreamExt;
+/// use quinn_h3::{
+///     server::{Connecting, IncomingRequest},
+///     Error,
+/// };
+///
+/// async fn handle_connection(connecting: Connecting) -> Result<(), Error> {
+///     let mut incoming_request = connecting.await?;
+///
+///     while let Some(recv_request) = incoming_request.next().await {
+///         let (request, _, _) = recv_request.await?;
+///         println!("Received request: {:?}", request);
+///     }
+///
+///     Ok(())
+/// }
+/// ```
+///
+/// [`RecvRequest`]: struct.RecvRequest.html
 pub struct IncomingRequest(ConnectionRef);
 
 impl IncomingRequest {
+    /// Gracefully shutdown the connection.
+    ///
+    /// All currently running connection will be honored, as well as those sent before the
+    /// client receieved the GoAway frame.
```

Comment:
```suggestion
    /// All currently running connection will be honored, as well as those sent before the
    /// client received the GoAway frame.
```

I'm confused about the notion of "connection" here -- since this is only about a single request?

Reply:
Yeah, it's "requests", not "connections". Thanks !